grassroot-webapp/src/main/java/za/org/grassroot/webapp/util/USSDEventUtil.java

```diff
@@ -190,24 +191,42 @@ public void updateExistingEvent(String userUid, String requestUid, String fieldC
      * Method that invokes the date time parser directly, with no attempt to map to a predefined format or regex. Should
      * likely move to a single class in language, once updated that module to Java 8 and Antlr 4
      * @param passedValue
+     * @param eventUid
      * @return LocalDateTime of most likely match; if no match, returns the current date time rounded up to next hour
      */
-    public LocalDateTime parseDateTime(String passedValue) throws SeloParseDateTimeFailure {
+    public LocalDateTime parseDateTime(String passedValue, String eventUid) throws SeloParseDateTimeFailure {
 
         LocalDateTime parsedDateTime = DateTimeUtil.tryParseString(passedValue);
 
+        log.info("Local Date Time processed******************{}",parsedDateTime);
+
+        String dateTime = "";
+
         if (parsedDateTime == null) {
             try {
                 parsedDateTime = learningService.parse(passedValue);
                 log.info("Date time processed: " + parsedDateTime.toString());
+                dateTime = parsedDateTime.toString();
             } catch (Exception e) {
                 throw new SeloParseDateTimeFailure();
             }
         }
 
-        return parsedDateTime;
+        LocalDateTime localDateTime;
+
+        if(dateTime.substring(dateTime.indexOf("T") + 1).equals("00:00")){
```

Comment:
This is fragile, e.g., to changes in how to string works. ParsedDateTime is a time entity, you can extract the hour and minute.